Switch to new focus brushes

chingucoding opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->
Switching to new focus brushes.
## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->
<!--- Use the syntax "Closes #1234" or "Fixes #5678" so that GitHub will close the issue once the PR is complete. -->
Closes #4560 
## How Has This Been Tested?
<!--- Please describe how you tested your changes. -->
Tested manually.
## Screenshots (if appropriate):
<!--- If you are making visual changes to a Control or adding a new Control, please include screenshots showing the result. -->

![Screenshot of focuses AutosuggestBox with white focus rect in dark theme](https://user-images.githubusercontent.com/16122379/111595171-25ab6600-87cc-11eb-9a41-a41e42fa93ba.png)
![Screenshot of focuses AutosuggestBox with black focus rect in light theme](https://user-images.githubusercontent.com/16122379/111595234-352aaf00-87cc-11eb-89f4-f07316503723.png)

ranjeshj commented:
@marksfoster @YuliKl Is the design intent to change these targeted backgrounds or to update the system focus rect colors ?

ranjeshj commented on dev/CommonStyles/CalendarDatePicker_themeresources_v1.xaml:
```diff
@@ -22,7 +22,7 @@
             <StaticResource x:Key="CalendarDatePickerBackgroundPointerOver" ResourceKey="SystemControlPageBackgroundAltMediumBrush" />
             <StaticResource x:Key="CalendarDatePickerBackgroundPressed" ResourceKey="SystemControlBackgroundBaseLowBrush" />
             <StaticResource x:Key="CalendarDatePickerBackgroundDisabled" ResourceKey="SystemControlBackgroundBaseLowBrush" />
-            <StaticResource x:Key="CalendarDatePickerBackgroundFocused" ResourceKey="SystemControlHighlightListAccentLowBrush" />
```
This change should only be for the new styles where we are using the new colors.

chingucoding replied:
Fixed now.

YuliKl commented:
I apologize, I was too vague when originally creating this work item internally. The intent is to update two brushes from generic.xaml, `SystemControlFocusVisualPrimaryBrush` and `SystemControlFocusVisualSecondaryBrush` to use new colors: `FocusStrokeColorOuter` and `FocusStrokeColorInner`.
https://github.com/microsoft/microsoft-ui-xaml/blob/d2f08547d2a4c4a1b8be251811aa10d0640759c4/dev/CommonStyles/Common_themeresources_any.xaml#L77

chingucoding commented:
@YuliKl So we would introduce those two brushes in the WinUI code base now and point them to those two resources?

chingucoding commented:
Updated that now.

YuliKl commented on dev/CommonStyles/Common_themeresources_any.xaml:
```diff
@@ -564,6 +571,9 @@
             <SolidColorBrush x:Key="SystemFillColorCriticalBackgroundBrush" Color="{ThemeResource SystemColorWindowColor}" />
             <SolidColorBrush x:Key="SystemFillColorNeutralBackgroundBrush" Color="{ThemeResource SystemColorWindowColor}" />
 
+            <SolidColorBrush x:Key="SystemControlFocusVisualPrimaryBrush" Color="{StaticResource SystemColorButtonTextColor}"/>
```
While you're here, @chingucoding, would you mind updating this to SystemColorWindowTextColor? (We have another issue tracking that request)
```suggestion
            <SolidColorBrush x:Key="SystemControlFocusVisualPrimaryBrush" Color="{StaticResource SystemColorWindowTextColor }"/>
```

chingucoding replied:
Sure! Only Primary brush @YuliKl ?

YuliKl commented on dev/CommonStyles/Common_themeresources_any.xaml:
```diff
@@ -564,6 +571,9 @@
             <SolidColorBrush x:Key="SystemFillColorNeutralBackgroundBrush" Color="{ThemeResource SystemColorWindowColor}" />
 
+            <SolidColorBrush x:Key="SystemControlFocusVisualPrimaryBrush" Color="{StaticResource SystemColorButtonTextColor}"/>
+            <SolidColorBrush x:Key="SystemControlFocusVisualSecondaryBrush" Color="{StaticResource SystemColorButtonTextColor}"/>
```
And this one should be a matching pair to WindowTextColor - so WindowColor.
```suggestion
            <SolidColorBrush x:Key="SystemControlFocusVisualSecondaryBrush" Color="{StaticResource SystemColorWindowColor}"/>
```

ranjeshj commented on dev/CommonStyles/Common_themeresources_any.xaml:
```diff
@@ -564,6 +571,9 @@
             <SolidColorBrush x:Key="SystemFillColorCriticalBackgroundBrush" Color="{ThemeResource SystemColorWindowColor}" />
             <SolidColorBrush x:Key="SystemFillColorNeutralBackgroundBrush" Color="{ThemeResource SystemColorWindowColor}" />
 
+            <SolidColorBrush x:Key="SystemControlFocusVisualPrimaryBrush" Color="{StaticResource SystemColorWindowTextColor }"/>
```
You should move these changes into Common_themeresources.xaml. The _any is included in both version1 and version2 styles.

chingucoding replied:
Done now.

ranjeshj commented:
/azp run

StephenLPeters commented:
@chingucoding do these visual verification failure makes sense?

chingucoding commented:
Failures looked good to me, we changed quite a core brush that is referenced by multiple controls. most importantly ListView which is used in other controls. Updated the verification files now.

ranjeshj commented:
/azp run

ranjeshj commented:
@chingucoding looks like a few failures in rs4 where the baselines dont match.

chingucoding commented:
Updated them now.

ranjeshj commented:
/azp run

chingucoding commented on dev/CommonStyles/Common_themeresources.xaml:
```diff
@@ -17,6 +17,8 @@
             <SolidColorBrush x:Key="ApplicationPageBackgroundThemeBrush" Color="{StaticResource SolidBackgroundFillColorBase}" />
             <StaticResource x:Key="DefaultTextForegroundThemeBrush" ResourceKey="TextFillColorPrimaryBrush" />
+            <SolidColorBrush x:Key="SystemControlFocusVisualPrimaryBrush" Color="{StaticResource FocusStrokeColorOuter}"/>
+            <SolidColorBrush x:Key="SystemControlFocusVisualSecondaryBrush" Color="{StaticResource FocusStrokeColorInner}"/>
```
Would you like to incorporate those changes in one of the existing PRs (e.g. #4628) or should we create a new PR to fix this?

marksfoster replied:
Would you be able to add it to your open PR: 4635?

Sorry these comments are late. I couldn't figure out why nobody was responding to any of my PR comments and found out you need to publish your comments so others can see them. Confusing UI!

chingucoding replied:
No problem, I'll add it to #4635 :)
In case you didn't know there are actually two different ways to comment, one posts them immediately the other waits for them to be published as part of a review.
